src/main/java/BasicMAPF/Solvers/PrioritisedPlanning/PrioritisedPlanning_Solver.java

```diff
@@ -336,12 +334,23 @@ private static InstanceReport initSubproblemReport(MAPF_Instance instance) {
 
     protected RunParameters getSubproblemParameters(MAPF_Instance subproblem, InstanceReport subproblemReport, ConstraintSet constraints, float maxCost, Solution solutionSoFar) {
         long timeLeftToTimeout = Math.max(super.maximumRuntime - (System.nanoTime()/1000000 - super.startTime), 0);
-        RunParameters_SAAStar params = new RunParameters_SAAStar(new RunParametersBuilder().setTimeout(timeLeftToTimeout).
-                setConstraints(new ImmutableConstraintSet(constraints)).setInstanceReport(subproblemReport).setAStarGAndH(this.singleAgentGAndH).createRP());
+
+        RunParameters_SAAStar params;
+        if (transientMAPFBehaviour == TransientMAPFBehaviour.transientMAPFsst) {
+            params = new RunParameters_SAAStar(new RunParametersBuilder().setTimeout(timeLeftToTimeout).
+                    setConstraints(new ImmutableConstraintSet(constraints)).setInstanceReport(subproblemReport).setAStarGAndH(new TransientDistanceTableSingleAgentHeuristic(this.agents, subproblem.map)).createRP());
+        }
+        else {
+            params = new RunParameters_SAAStar(new RunParametersBuilder().setTimeout(timeLeftToTimeout).
+                    setConstraints(new ImmutableConstraintSet(constraints)).setInstanceReport(subproblemReport).setAStarGAndH(this.singleAgentGAndH).createRP());
+        }
+
```

Comment:
you're rebuilding the heuristic from scratch before each agent, which is what's causing this to run so slowly. Once you've changed the implementation to a composition this will be solved. 
Also, it will be cleaner if you used a ternary expression for this one argument rather than duplicate the whole thing twice.